services/skus/credentials.go

```diff
@@ -32,169 +33,162 @@ const (
 	defaultCohort             int16 = 1
 )
 
-// Retry and backoff variables
 var (
-	defaultExpiresAt   = time.Now().Add(17532 * time.Hour) // 2 years
-	retryPolicy        = retrypolicy.DefaultRetry
-	nonRetriableErrors = []int{http.StatusBadRequest, http.StatusUnauthorized, http.StatusForbidden,
-		http.StatusInternalServerError, http.StatusConflict}
-)
-
-var (
-	// ErrOrderUnpaid - unpaid order variable
-	ErrOrderUnpaid = errors.New("order not paid")
-
-	// ErrOrderHasNoItems is an order has no order items error
+	ErrOrderUnpaid     = errors.New("order not paid")
 	ErrOrderHasNoItems = errors.New("order has no items")
-)
 
-// Issuer includes information about a particular credential issuer
-type Issuer struct {
-	ID         uuid.UUID `json:"id" db:"id"`
-	CreatedAt  time.Time `json:"createdAt" db:"created_at"`
-	MerchantID string    `json:"merchantId" db:"merchant_id"`
-	PublicKey  string    `json:"publicKey" db:"public_key"`
-}
+	errInvalidIssuerResp model.Error = "invalid issuer response"
 
-// Name returns the name of the issuer as known by the challenge bypass server
-func (issuer *Issuer) Name() string {
-	return issuer.MerchantID
-}
+	defaultExpiresAt = time.Now().Add(17532 * time.Hour) // 2 years
+	retryPolicy      = retrypolicy.DefaultRetry
+	dontRetryCodes   = map[int]struct{}{
+		http.StatusBadRequest:          struct{}{},
+		http.StatusUnauthorized:        struct{}{},
+		http.StatusForbidden:           struct{}{},
+		http.StatusInternalServerError: struct{}{},
+		http.StatusConflict:            struct{}{},
+	}
+)
 
-// CreateIssuer creates a new v1 issuer if it does not exist. This only happens in the event of a new sku being created.
-func (s *Service) CreateIssuer(ctx context.Context, merchantID string, orderItem OrderItem) error {
-	issuerID, err := encodeIssuerID(merchantID, orderItem.SKU)
+// CreateIssuer creates a new v1 issuer if it does not exist.
+//
+// This only happens in the event of a new sku being created.
+func (s *Service) CreateIssuer(ctx context.Context, dbi sqlx.QueryerContext, merchID string, item *OrderItem) error {
+	encMerchID, err := encodeIssuerID(merchID, item.SKU)
 	if err != nil {
 		return errorutils.Wrap(err, "error encoding issuer name")
 	}
 
-	issuer, err := s.Datastore.GetIssuer(issuerID)
-	if err != nil && !errors.Is(err, sql.ErrNoRows) {
-		return fmt.Errorf("error get issuer for issuerID %s: %w", issuerID, err)
+	_, err = s.issuerRepo.GetByMerchID(ctx, dbi, encMerchID)
+	// Found, nothing to do.
+	if err == nil {
+		return nil
 	}
 
-	if issuer == nil {
-		logging.FromContext(ctx).Info().
-			Msgf("creating new issuer %s", issuerID)
+	if !errors.Is(err, model.ErrIssuerNotFound) {
+		return fmt.Errorf("error get issuer for issuerID %s: %w", encMerchID, err)
+	}
 
-		requestOperation := func() (interface{}, error) {
-			return nil, s.cbClient.CreateIssuer(ctx, issuerID, defaultMaxTokensPerIssuer)
-		}
+	logging.FromContext(ctx).Info().Msgf("creating new issuer %s", encMerchID)
```

Comment:
Do you think we need this? Just thinking about trying to reduce noise in our logs and not sure how much value this adds.

Reply:
I don't think we need this. I did not add it – just slightly moved. Happy to delete it. 🙂

Reply:
Deleted this line as well as the same line from `CreateIssuerV3`.